### ISSUE TITLE
Could CaffeineCache expose "AsyncCache<> cache" field or at least its asMap method?

### ISSUE THREAD

**Issue by jean-merelis** [quarkusio/quarkus#6903]
I have a cache with a composite key (userId, companyId) and I need to invalidate all entries only by companyId.
I can't currently do that. I have to invalidate the entire cache with `CacheInvalidateAll()`
I think with `AsyncCache.asMap()` I can iterate over it and get what I want

**Comment by jean-merelis** [comment]
I think the DefaultCacheKey and CompositeCacheKey classes should be public as well. This way I could interact programmatically with the cache.

**Comment by jean-merelis** [comment]
I think the DefaultCacheKey and CompositeCacheKey classes should be public as well. This way I could interact programmatically with the cache.

**Comment by gsmet** [comment]
I wonder if it would be better to make them distinct caches and have an annotation that can add elements to the cache name.

@ben-manes @gwenneg WDYT?

**Comment by gwenneg** [comment]
This sounds like a feature of the API for the more advanced use cases that was discussed in #3306 (it's supposed to be part of the MVP). I think we'll need the opinion of @emmanuelbernard, @Sanne or @galderz on this.

**Comment by emmanuelbernard** [comment]
right, let's add this requirement to the programmatic API work we want to do. Maybe referencing this issue from a uber issue.

**Comment by emmanuelbernard** [comment]
I anticipate we would offer Mutiny non blocking types and such instead of what Caffeine offers today.
